client/app/actions/commentsActionCreators.js

```diff
@@ -43,24 +45,34 @@ export function submitCommentFailure(error) {
 
 export function fetchComments() {
   return dispatch => {
+    dispatch(setIsFetching());
     return (
-      commentsManager
-        .fetchComments()
-        .then(res => dispatch(fetchCommentsSuccess(res.data)))
-        .catch(res => dispatch(fetchCommentsFailure(res.data)))
+        commentsManager
+            .fetchComments()
+            .then(res => dispatch(fetchCommentsSuccess(res.data)))
+            .catch(res => dispatch(fetchCommentsFailure(res.data)))
     );
   };
 }
 
 export function submitComment(comment) {
+  const startTime = (new Date()).getTime();
   return dispatch => {
-    dispatch(incrementAjaxCounter());
+    dispatch(setIsSaving());
     return (
       commentsManager
         .submitComment(comment)
-        .then(res => dispatch(submitCommentSuccess(res.data)))
+        .then(res => {
+          const result = res.data;
+          const successTime = (new Date()).getTime();
+          const timeDiff = successTime - startTime;
+          if (timeDiff < minimumDelay) {
+            setTimeout(dispatch, minimumDelay - timeDiff, submitCommentSuccess(result));
+          } else {
+            dispatch(submitCommentSuccess(result));
+          }
+        })
         .catch(res => dispatch(submitCommentFailure(res.data)))
```

Comment:
The idea of minimum delay is just for UX.
If the promise return very fast, you just have a blink, and it's kind of confusing.
But I agree that it's too much code for a starter repo...